torch/testing/_internal/common_methods_invocations.py

```diff
@@ -18888,6 +18907,20 @@ def __init__(
             }), 'TestCommon', 'test_python_reference_consistency', device_type='cpu'),
         ),
     ),
+    PythonRefInfo(
```

Comment:
How thorough are the sample inputs for these operations? Should we look at adding reference inputs?

Reply:
I think since they are composite ops (even for torch implementation), the sample inputs check basic cases and should be sufficient assuming the primitive ops are tested well. Wdyt?

Also I have updated `margin_ranking_loss` to pass reduction (there was a TODO for the same on sample_inputs).

Reply:
It's true that the more primitive operations are more tested, but the combination of those operations to create these operators is novel, so it never hurts to stress-test edge cases by adding some reference inputs

Reply:
Have added referencce_input_func for both of them. PTAL :)